addons/payment/static/src/js/manage_form.js

```diff
@@ -80,26 +80,19 @@ odoo.define('payment.manage_form', require => {
         _deleteToken: function (tokenId) {
             const execute = () => {
                 this._rpc({
-                    model: 'payment.token',
-                    method: 'write',
-                    args: [[tokenId], {active: false}],
+                    route: '/my/payment_method/archive',
+                    params: {
+                        'token_id': tokenId,
+                    },
                 }).then(result => {
-                    if (result === true) { // Token successfully deleted, remove it from the view
-                        const $tokenCard = this.$(
-                            `input[name="o_payment_radio"][data-payment-option-id="${tokenId}"]` +
-                            `[data-payment-option-type="token"]`
-                        ).closest('div[name="o_payment_option_card"]');
-                        $tokenCard.siblings(`#o_payment_token_inline_form_${tokenId}`).remove();
-                        $tokenCard.remove();
-                        this._disableButton(false);
-                    }
-                }).guardedCatch(error => {
-                    this._displayError(
-                        _t("Server Error"),
-                        _t("We are not able to delete your payment method."),
-                        error.message.data.message
-                    );
-                });
```

Comment:
You must keep the error processing because some acquirers request their provider to delete the token on their side. If this fails, an error message should be shown rather than hiding the token that would be back after a refresh. Since the `result` was returned by the RPC to `write`, you must return it from the controller you added (nothing too fancy, just pass it to the JS as-is).

Reply:
Hello @AntoineVDV :slightly_smiling_face:  , 
Thanks for your review! Everything has been changed according to your feedback.

I'm just not sure about this point because you asked me to change `token_sudo.write({'active': False})` to `token_sudo.active = False` but now I'm using the return of `return token_sudo.write({'active': False})` like before. 

Is there anything better to do? :thinking: 
Thanks!

Reply:
As stated in my last comment, the last part of my request was bullshit and you immediately can see it when you write down `return token_sudo.active = False` and `rtype: bool|Non` :sweat_smile: